File: example/basicjava/1-simple/test/src/FooTest.java
package foo;
import static org.junit.Assert.assertEquals;
import org.junit.Test;

public class FooTest {
    @Test
    public void testSimple() {
        assertEquals(Foo.generateHtml("hello"),  "<h1>hello</h1>");
    }

    @Test
    public void testEscaping() {
        assertEquals(Foo.generateHtml("<hello>"),  "<h1>&lt;hello&gt;</h1>");
    }
}

File: example/kotlinlib/basic/1-script/FooTest.kt
//| extends: [mill.script.KotlinModule.Junit5]
//| moduleDeps: [./Foo.kt]
//| mvnDeps:
//| - "io.kotest:kotest-runner-junit5:5.9.1"
//| - "com.github.sbt.junit:jupiter-interface:0.11.2"

import io.kotest.core.spec.style.FunSpec
import io.kotest.matchers.shouldBe

class FooTest :
    FunSpec({
        test("testSimple") {
            generateHtml("hello") shouldBe "<h1>hello</h1>"
        }

        test("testEscaping") {
            generateHtml("<hello>") shouldBe "<h1>&lt;hello&gt;</h1>"
        }
    })
